### PR TITLE
feat: disallow collapsing sidebar when signposting is enabled

### PR DESCRIPTION
**Pull request by akash-codemonk**
## Description

We hide the sidebar collapse button during signposting. 
We reset sidebar pinned state to show by default when an app is reopened.

#### PR fixes following issue(s)
Fixes https://github.com/appsmithorg/appsmith/issues/24345

#### Type of change

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

#### How Has This Been Tested?

- [x] Manual
- [x] Jest
- [x] Cypress

#### Test Plan
https://github.com/appsmithorg/appsmith/pull/24938#issuecomment-1619972894

#### Issues raised during DP testing
https://github.com/appsmithorg/appsmith/pull/24938#issuecomment-1620087603
https://github.com/appsmithorg/appsmith/pull/24938#issuecomment-1623447074

#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [x] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [x] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [x] Test plan has been peer reviewed by project stakeholders and other QA members
- [x] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [x] Cypress test cases have been added and approved by SDET/manual QA
- [x] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

**Comment by kavitasmoolya** [comment]
@akash-codemonk Above DP is down

**Comment by kavitasmoolya** [comment]
**Scenarios to test**

- [ ] Sidebar cannot be collapsed by the user when signposting is enabled. The icon will be hidden

- [ ] When signposting is not enabled, the user can collapse the sidebar, the icon will be visible

- [ ] Today on release/prod the sidebar will be closed on reopens/edit/creating a new app. With this PR when user collapses the sidebar, exits the app, then if the user reopens/edit/creates a new app the sidebar should be open by default i.e by default the sidebar will be open when a user opens an app.

**Comment by kavitasmoolya** [comment]
- [ ] Ctrl+/ is hiding the sidebar due to which canvas size is not fixed
image

**Comment by kavitasmoolya** [comment]
> **Issues :**
> 
> * [x]  Ctrl+/ is hiding the sidebar due to which canvas size is not fixed
>   image

Fix is working fine on DP and verified successfully
image

**Comment by akash-codemonk** [comment]
> * [ ]  Sidebar collapses on clicking app setting option in edit mode.

Expected not an issue. Since it is not a user action. 

> * [ ]  Side bar icon displays for a very short time on navigating from home page to app edit mode
>   https://www.loom.com/share/83016cdb310d4334a02e3d0de4b2119b?sid=fb8d7041-ac14-48c0-a46e-cddd7eb894cc

It will take some time to load info as to whether to start signposting or not. I feel it is a low priority. No fix for now. 

> * [ ]  user collapses the sidebar, preview the app, then if the user comes back to app edit mode the sidebar is not open by default
>   https://www.loom.com/share/78ba1374820e4ad9aaf038a18c515319?sid=82e0bfc1-5e88-444b-9b09-d5821d91696c

No fix. This is expected behaviour

@kavitasmoolya

File: app/client/src/sagas/InitSagas.ts [hunk 1]
```diff
@@ -109,6 +113,8 @@ function* resetEditorSaga() {
   // previously
   yield put(setPreviewModeAction(false));
   yield put(resetSnipingMode());
+  yield put(setExplorerActiveAction(true));
```
Comment (hetunandu) [inline app/client/src/sagas/InitSagas.ts:116]: Why should we always reset this?

Comment (akash-codemonk) [reply app/client/src/sagas/InitSagas.ts:116]: User might have collapsed the sidebar and exited the app. The state remains as it is when a new app is created or edited. We intend to show the sidebar by default when an app is reopened or created. 

This was mentioned by rahul. Some users seem to be confused as to what to do/stuck when the sidebar is closed, and they tend to try to create a new app post that.

**Comment by kavitasmoolya** [comment]
@akash-codemonk 

- [ ] Side bar is not opened by default on changing from main to any other branch for git connected app
[screen-capture (5).webm](https://github.com/appsmithorg/appsmith/assets/134914630/4dec403c-ad68-42bc-8791-d41b64134a65)

**Comment by akash-codemonk** [comment]
> * screen-capture (5).webm

Expected, when the user inside the app the user preference stays. It is set to default when they initially edit the application.

**Review by kavitasmoolya — DISMISSED** [review]
Testing is completed on DP and no open issues.

**Review by hetunandu — APPROVED** [review]

**Pull request merged**